### PR TITLE
NEW: etna.auto.pool module

### LINKED ISSUE #859: `etna.auto.pool` module
### 🚀 Feature Request

Module and classes for config Pools handling:

ToBe:
- Module with default template used for model choice
- We should have general method for classes initialization with given parameters like horizon
- enum Pool for easy pool choice
- Use omegaconf for template filling via `Omegaconf.resolve`

### Proposal

```python

class PoolGenerator:
     def __init__(self, configs_template: List[dict]):
        pass

     def generate(self, horizon: int) -> List[Pipeline]:
          "fill templates with horizon value"

class Pool(Enum):
    default = PoolGenerator(configs_template=pool.templates.DEFAULT)
```

```python
# etna.auto.pool.templates

DEFAULT = [
{"__target__": "etna.pipeline.Pipeline", "horizon": "${horizon}" , ...},
#TBA,  
#TBA, 
...
]

```

### Test cases

- Check if default pool properly instantiate
- PoolGenerator for custom list templates
- Check if `pool.templates.DEFAULT` not changed after multiple generations

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by martins0n**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #859

### CODE REVIEW

File: etna/auto/pool/templates.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # {'_target_': 'etna.pipeline.Pipeline', 'horizon': '${__aux__.horizon}', 'model': {'_target_': 'etna.models.CatBoostMultiSegmentModel'}, 'transforms': [{'_target_': 'etna.transforms.TimeSeriesImputerTransform', 'in_column': 'target', 'strategy': 'mean'}, {'_target_': 'etna.transforms.LogTransform', 'in_column': 'target'}, {'_target_': 'etna.transforms.BinsegTrendTransform', 'in_column': 'target', 'jump': 1, 'min_size': 10}, {'_target_': 'etna.transforms.SegmentEncoderTransform'}, {'_target_': 'etna.transforms.LagTransform', 'in_column': 'target', 'lags': '${shift:${horizon},[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]}'}, {'_target_': 'etna.transforms.DateFlagsTransform', 'day_number_in_month': True, 'day_number_in_week': True, 'is_weekend': True, 'month_number_in_year': True, 'week_number_in_month': True, 'week_number_in_year': True, 'year_number': True}]} ,
```
Comment (Mr-Geekman) [inline etna/auto/pool/templates.py:6]: What happened to these commented lines?

**Comment by Mr-Geekman** [comment]
I can't find a `auto` module in documentation and check formatting. We should probably add it somewhere in docs.

File: etna/auto/pool/templates.py (hunk 1 above)
Comment (martins0n) [reply etna/auto/pool/templates.py]: I'am going to add extra task for add pipelines to templates

**Comment by martins0n** [comment]
I've added docs there

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**